### PR TITLE
Doc: Release-notes v3.5.0 

### PR DESCRIPTION
**Pull request by emilarista**
## Change Summary

Adds release notes for 3.5.0

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/docs/release-notes/3.x.x.md [hunk 1]
```diff
@@ -1,8 +1,131 @@
 # Release Notes For Ansible AVD 3.x.x
 
+Documentation for AVD version `3.5.0` [available here](https://www.avd.sh/en/v3.5.0/)
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/docs/release-notes/3.x.x.md]: we have a catch-22 here. We cannot create this URL without the release, and we cannot create the release without RN pointing to this url.

I suggest to change the 3.5 to point to `/en/stable` and then we change to v3.5.0 when we create v3.6.0

Comment (carlbuchmann) [reply ansible_collections/arista/avd/docs/release-notes/3.x.x.md]: I like that idea, since it is accurate, once when release it!

**Comment by ClausHolbechArista** [comment]
I have removed CI and refactoring PRs from RN. Added `- ` before each PR. Reordered a few others.

**Review by ClausHolbechArista — DISMISSED** [review]
LGTM

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**